api/end_device.proto

```diff
@@ -303,16 +303,16 @@ message MACState {
 
   message JoinRequest {
     reserved 1,2,3,4,5,9,10; // deprecated JoinRequest fields.
-    DLSettings downlink_settings = 6 [(gogoproto.nullable) = false, (validate.rules).message.required = true];
+    DLSettings downlink_settings = 6 [(validate.rules).message.required = true];
     RxDelay rx_delay = 7 [(validate.rules).enum.defined_only = true];
     CFList cf_list = 8;
   }
   message JoinAccept {
     // Payload of the join-accept received from Join Server.
     bytes payload = 1 [(validate.rules).bytes = {min_len: 17, max_len: 33}];
-    JoinRequest request = 2 [(gogoproto.nullable) = false];
+    JoinRequest request = 2;
```

Comment:
I think this one can (and should) be required.

NS doesn't do any `nil` checking on it (only usage is `queued_join_accept` in ED).

Afaik we don't really accept this message through API anywhere, but if we ever do, that would be clean.